tests/entities_column_value_complex_tests.py

```diff
@@ -117,4 +118,113 @@ def test_should_item_link_column_with_api_data_with_no_linkedpulseid_key():
 
 
     # Assert 
-    eq_(column_value.value,[])
\ No newline at end of file
+    eq_(column_value.value,[])
+
+def test_should_week_column_with_no_api_data():
+
+    # Arrange
+    id = 'week1'
+    title="New Week"
+    column_type = ColumnType.week
+
+    # Act
+    column_value = en.cv.create_column_value(column_type,id=id,title=title)
+    format = column_value.format()
+
+    # Assert 
+    eq_(format,{})
+
+def test_should_week_column_with_api_data():
+
+    # Arrange
+    id = 'week1'
+    title="New Week"
+    column_type = ColumnType.week
+    api_return_value = {
+                        'startDate': '2021-09-20',
+                        'endDate': '2021-09-26'
+                        }
+    value = json.dumps(api_return_value)
+    column_value = en.cv.create_column_value(column_type,id=id,title=title,value=value)
+
+    # Act
+    format = column_value.format()
+
+    # Assert 
+    eq_(format['startDate'],'2021-09-20')
+
+def test_should_set_none_value_to_week_column_value():
+    id = 'week1'
+    title="New Week"
+    column_type = ColumnType.week
+    value = None
+    column_value = en.cv.create_column_value(column_type,id=id,title=title)
+    # Act
+    column_value.value = value
+
+    # Assert 
+    eq_(column_value.value,value)
+
+def test_should_test_week_value_to_week_column_value():
+    id = 'week1'
+    title="New Week"
+    column_type = ColumnType.week
+    startDate = datetime(year=2021, month=9, day=27)
+    endDate= datetime(year=2021, month=10, day=3)
+    value = Week(start=startDate,end=endDate)
+    column_value = en.cv.create_column_value(column_type,id=id,title=title)
+
+    # Act
+    column_value.value = value
+    
+    # Assert
+    eq_(column_value.value.start, value.start)
+    eq_(column_value.value.end, value.end)
+
+def test_should_set_dict_value_to_week_column_value():
+    id = 'week1'
+    title="New Week"
+    column_type = ColumnType.week
+    value =   {
+                'start': datetime(year=2021, month=9, day=27),
+                'end': datetime(year=2021, month=10, day=3)
+            }
+    startDate = datetime(year=2021, month=9, day=27)
+    endDate= datetime(year=2021, month=10, day=3)
+    week = Week(start=startDate,end=endDate)
+    column_value = en.cv.create_column_value(column_type,id=id,title=title)
+
+    # Act
+    column_value.value = value
+    
+    # Assert
+    eq_(column_value.value.start, week.start)
+    eq_(column_value.value.end, week.end)
+
+@raises(e.ColumnValueError)
+def test_should_set_invalid_dict_value_to_week_column_value():
+    id = 'week1'
+    title="New Week"
+    column_type = ColumnType.week
+    value =   { 'this': 'dictionary is invalid'}
+    column_value = en.cv.create_column_value(column_type,id=id,title=title)
+
+    # Act
+    column_value.value = value 
+
+def test_should_set_none_start_value_to_week_column_value():
+    id = 'week1'
+    title="New Week"
+    column_type = ColumnType.week
+    api_return_value =   {
```

Comment:
You do not need to set initial API data for this one.  Simply create a Week object, set one of the proprties (start or end) to None, and pass it into _column_value.value_.